core/backend/src/workspace/Settings.ts

```diff
@@ -261,62 +287,71 @@ export class BaseSettings implements Settings {
     return false;
   }
 
-  public resolveSetting<T extends SettingType>(name: SettingName, resolver: SettingResolver<T>, defaultValue?: T): T | undefined {
-    for (const dict of this._dictionaries) {
-      const val = dict.getSetting(name) as T | undefined;
-      const resolved = val && resolver(val, dict.name, dict.priority);
-      if (undefined !== resolved) {
+  public resolveSetting<T extends SettingType>(arg: { settingName: SettingName, resolver: Settings.Resolver<T> }, defaultValue?: T): T | undefined {
+    for (const dict of this.dictionaries) {
+      const val = dict.getSetting<T>(arg.settingName);
+      const resolved = val && arg.resolver(val, dict);
+      if (undefined !== resolved)
         return resolved;
-      }
     }
     return defaultValue;
   }
 
-  public getSetting<T extends SettingType>(name: SettingName, defaultValue?: T): T | undefined {
-    return this.resolveSetting(name, (val) => deepClone<T>(val)) ?? defaultValue;
+  public getSetting<T extends SettingType>(settingName: SettingName, defaultValue?: T): T | undefined {
+    return this.resolveSetting({ settingName, resolver: (val) => deepClone<T>(val) }) ?? defaultValue;
   }
 
   /** for debugging. Returns an array of all values for a setting, sorted by priority.
    * @note values are not cloned. Do not modify objects or arrays.
    */
-  public inspectSetting<T extends SettingType>(name: SettingName): SettingInspector<T>[] {
-    const all: SettingInspector<T>[] = [];
-    this.resolveSetting<T>(name, (value, dictionary, priority) => {
-      all.push({ value, dictionary, priority });
-      return undefined;
+  public inspectSetting<T extends SettingType>(settingName: SettingName): Settings.Inspector<T>[] {
+    const all: Settings.Inspector<T>[] = [];
+    this.resolveSetting<T>({
+      settingName, resolver: (value, dictionary) => {
+        all.push({ value, dictionary });
+        return undefined;
+      },
     });
 
     return all;
   }
 
+  // get the setting and verify the result is either undefined or the correct type. If so, return it. Otherwise throw an exception.
+  private getResult<T extends SettingType>(name: SettingName, expectedType: string) {
```

Comment:
This is a change to the error handling of the "getXXX" methods for Settings. Previously settings were not validated against the SettingSchema when their values were returned. If you called `getString`, and the found value was a string it was returned, regardless of whether the schema indicated it should be a string. Also, if you called `getString` and the result was a type other than `string` it returned the default value. Now, settings are always validated against the schema (if one exists) and an exception is thrown if the value is not the expected type. This change affected several tests.